python/pyarrow/array.pxd

```diff
@@ -109,7 +109,9 @@ cdef class BinaryArray(Array):
 
 
 cdef class DictionaryArray(Array):
-    pass
+    cdef:
+        object _indices, _dictionary
```

Comment:
Do we really need to define them? As they are Python attributes, it should be ok to omit them?